File: observer/java/Store.java
// Store class extends ConcretePublisher
public class Store extends ConcretePublisher {

    // Constructor calls super constructor with given events
    public Store(String... events) {
        super(events);
    }

    // Method to update subscribers about tech products
    public void updateTechProducts() {
        notify("tech products");
    }

    // Method to update subscribers about sports products
    public void updateSportsProducts() {
        notify("sports products");
    }

}

File: observer/TypeScript/Store.ts
import ConcretePublisher from "./ConcretePublisher";

export default class Store extends ConcretePublisher {
  // Constructor calls super constructor with given events
  constructor(events: string[]) {
    super(events);
  }

  // Method to update subscribers about tech products
  updateTechProducts(): void {
    this.notify("tech products");
  }

  // Method to update subscribers about sports products
  updateSportsProducts(): void {
    this.notify("sports products");
  }
}
